src/gui/FxMixerView.cpp

```diff
@@ -439,38 +436,32 @@ void FxMixerView::deleteUnusedChannels()
 	tracks += Engine::getSong()->tracks();
 	tracks += Engine::getBBTrackContainer()->tracks();
 
-	// go through all FX Channels
-	for(int i = m_fxChannelViews.size()-1; i > 0; --i)
+	bool inUse[m_fxChannelViews.size()] = {false};
```

Comment:
Variable length arrays aren't standard C++. Consider using a `std::vector` instead.

Reply:
What's variable about it?

Reply:
Ok, looked it up. Wow.

Reply:
`m_fxChannelViews.size()` isn't a constant expression - it can have different values at runtime. Standard C++ only allows arrays on the stack where the size is known at compile time.

Reply:
```suggestion
	std::vector<bool> inUse(m_fxChannelViews.size(), false);
```
Pretty sure that's all that's needed to change. Everything else is proper vector access.

Reply:
Oh, other than `#include <vector>`